src/storage/txn/process.rs

```diff
@@ -252,11 +267,24 @@ impl<E: Engine, S: MsgScheduler, L: LockManager> Executor<E, S, L> {
                         cid,
                         pr,
                         result: Ok(()),
+                        pipelined: false,
                         tag,
                     }
                 } else {
                     let sched = scheduler.clone();
                     let sched_pool = self.take_pool();
+                    let (write_finished_pr, pipelined_write_pr) = if pipelined {
+                        (
+                            if let Some(v) = pr.maybe_clone() {
```

Comment:
In fact, the `pr` must be `ProcessResult::PessimisticLockRes { Ok(PessimisticLockRes) }` here. It's clonable.